Source/DataSources/I3SDataSource.js

```diff
@@ -0,0 +1,4601 @@
+          featureData: that._featureData,
+          defaultGeometrySchema: that._layer._data.store.defaultGeometrySchema,
+          url: that._geometryData[0]._completeUri,
+          tile: that._tile,
+        });
+        task.then(function (data) {
+          rawGLTF = data.gltfData;
+          that._geometryData[0].customAttributes = data.customAttributes;
+          resolve();
+        });
+      } else {
+        resolve();
+      }
+    });
+
+    generateGLTF.then(function () {
+      var binaryGLTFData = that._dataSource._binarizeGLTF(rawGLTF);
+      var b3dmRawData = that._dataSource._binarizeB3DM(
+        featureTableJSON,
+        batchTableJSON,
+        binaryGLTFData
+      );
+      var b3dmDataBlob = new Blob([b3dmRawData], {
+        type: "application/binary",
+      });
+      that._b3dmURL = URL.createObjectURL(b3dmDataBlob);
+      resolve();
+    });
+  });
+};
+
+/**
+ * @private
+ */
+I3SNode.prototype._createVisualElements = function () {
+  if (!this._dataSource._traceVisuals) {
+    return;
+  }
+  var obb = this._data.obb;
+  var mbs = this._data.mbs;
+
+  if (obb) {
+    // Add an entity for display
+    var orientation = new Quaternion(
+      obb.quaternion[0],
+      obb.quaternion[1],
+      obb.quaternion[2],
+      obb.quaternion[3]
+    );
+
+    var obbPosition = _WGS84ToCartesian(
+      obb.center[0],
+      obb.center[1],
+      obb.center[2]
+    );
+
+    this._entities.locator = this._dataSource.entities.add({
+      name: "Extent",
+      position: obbPosition,
+      orientation: orientation,
+      box: {
+        dimensions: new Cartesian3(
+          obb.halfSize[0] * 2,
+          obb.halfSize[1] * 2,
+          obb.halfSize[2] * 2
+        ),
+        fill: false,
+        outline: true,
+        outlineColor: Color.ORANGE,
+      },
+    });
+  } else if (mbs) {
+    var mbsPosition = _WGS84ToCartesian(mbs[0], mbs[1], mbs[2]);
+
+    // Add an entity for display
+    this._entities.locator = this._dataSource.entities.add({
+      name: "Extent",
+      position: mbsPosition,
+      ellipse: {
+        semiMinorAxis: mbs[3],
+        semiMajorAxis: mbs[3],
+        fill: false,
+        outline: true,
+        outlineColor: Color.ORANGE,
+      },
+    });
+  }
+};
+
+/*
+.########.########....###....########.##.....##.########..########
+.##.......##.........##.##......##....##.....##.##.....##.##......
+.##.......##........##...##.....##....##.....##.##.....##.##......
+.######...######...##.....##....##....##.....##.########..######..
+.##.......##.......#########....##....##.....##.##...##...##......
+.##.......##.......##.....##....##....##.....##.##....##..##......
+.##.......########.##.....##....##.....#######..##.....##.########
+*/
+
+/**
+ * This class implements an I3S Feature
+ * @private
+ * @alias I3SFeature
+ * @constructor
+ * @param {I3SNode} [parent] The parent of that feature
+ * @param {String} [uri] The uri to load the data from
+ */
+function I3SFeature(parent, uri) {
```

Comment:
Like [`Cesium3DTileFeature`](https://github.com/CesiumGS/cesium/blob/main/Source/Scene/Cesium3DTileFeature.js), if this is the type that is returned when the user picks a feature, it should be marked as public.